modules/mesos/src/main/java/org/apache/ignite/mesos/resource/IgniteProvider.java

```diff
@@ -17,29 +17,52 @@
 
 package org.apache.ignite.mesos.resource;
 
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.BufferedReader;
 import java.io.File;
 import java.io.FileOutputStream;
 import java.io.IOException;
+import java.io.InputStreamReader;
 import java.net.HttpURLConnection;
 import java.net.URL;
 import java.nio.channels.Channels;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.nio.file.StandardCopyOption;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 import org.apache.ignite.mesos.ClusterProperties;
 
+import static org.apache.ignite.mesos.ClusterProperties.IGNITE_VERSION;
+
 /**
  * Class downloads and stores Ignite.
  */
 public class IgniteProvider {
-    // This constants are set by maven-ant-plugin.
-    /** */
-    private static final String DOWNLOAD_LINK = "http://ignite.run/download_ignite.php";
+    /** Logger. */
+    private static final Logger log = Logger.getLogger(IgniteProvider.class.getSimpleName());
 
+    // This constants are set by maven-ant-plugin.
     /** */
     private static final String DOWNLOAD_URL_PATTERN = "https://archive.apache.org/dist/ignite/%s/apache-ignite-fabric-%s-bin.zip";
 
+    /** URL for request Ignite latest version. */
+    private final static String IGNITE_LATEST_VERSION_URL = "https://ignite.apache.org/latest";
+
+    /** Mirrors. */
+    private static final String APACHE_MIRROR_URL = "https://www.apache.org/dyn/closer.cgi?as_json=1";
```

Comment:
I really don't like the duplication of this link here & in the pom.xml. Can we just have placeholders here for everything other than DOWNLOAD_URL_PATTERN? Also we don't need IGNITE_LATEST_VERSION_URL, we need *current* Ignite version. If we're on 2.5 we don't need to download 2.7. It should come from pom.xml.

Reply:
Agree on the duplication. Will remove it from the code (leaving only in pom.xml).

Sorry, what do you mean by "we need current Ignite version"? There's no such a thing...
At start, the framework will download the latest Ignite version by default. If a user needs a different version, he/she can specify it (and it'll be downloaded from the archives).